### PR TITLE
feat(rustc_unpretty): use rustc inspect mir

### PR DESCRIPTION
**Pull request by saying121**
Use `rustc -Z unpretty=mir` to inspect mir and other things,
and achieve an experience similar to Rust Playground.
The rust-analyzer view does not support viewing MIR for async blocks,
but rustc can do it.

### CODE REVIEW

**Comment by mrcjkb** [comment]
Hey :wave: 

This looks great! Thanks for the PR! :smile: 

I'll have a closer look at it later today.

**Review by mrcjkb — CHANGES_REQUESTED** [review]
Thanks again for this awesome PR! :heart: 

I have left some thoughts and suggestions.
Let me know if you'd like to implement them. If not, I'm happy to merge this into a feature branch and implement them myself :smile:

File: lua/rustaceanvim/commands/rustc_unpretty.lua [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  -- set filetype to rust for syntax highlighting
+  vim.bo[latest_buf_id].filetype = 'rust'
+  -- write the expansion content to the buffer
+  vim.api.nvim_buf_set_lines(latest_buf_id, 0, 0, false, lines)
+end
+
+---@param level rustcir_level
+function M.rustc_unpretty(level)
+  if #api.nvim_get_runtime_file('parser/rust.so', true) == 0 then
+    vim.notify('please install rust parser in nvim-treesitter', vim.log.levels.ERROR)
```
[the remark this answers is not in the document]
Comment (saying121) [reply lua/rustaceanvim/commands/rustc_unpretty.lua]: Is there a better way to obtain the function? 
I'm not sure what to do.

File: lua/rustaceanvim/commands/init.lua [hunk 2]
```diff
@@ -97,6 +97,27 @@ local command_tbl = {
     local cmd = args[1] or 'run'
     require('rustaceanvim.commands.fly_check')(cmd)
   end,
+  rustcUnpretty = function(args)
```
[the remark this answers is not in the document]
Comment (saying121) [reply lua/rustaceanvim/commands/init.lua:100]: I'm not familiar with the code structure, so I'll pass on this task.

**Comment by saying121** [comment]
I think I have done all I can do. 😄
Thanks for your work.

**Review by mrcjkb — APPROVED** [review]

**Comment by mrcjkb** [comment]
> I think I have done all I can do. 😄 Thanks for your work.

Thanks :pray:. I'll merge it into the `rustc-unpretty-continued` branch and continue working on it from there.

**Comment by mrcjkb** [comment]
@saying121 I've just finished the touchup and pushed a 4.1.0 release. The command is now `:Rustc unpretty`, and the config field is `tools.rustc`.

Thanks again for the awesome contribution :pray: :smile:

**Pull request merged**